src/mudlet.pro

```diff
@@ -103,6 +103,18 @@ isEmpty( BUILD ) {
 # Use APP_VERSION, APP_BUILD and APP_TARGET defines in the source code if needed.
 DEFINES += APP_VERSION=\\\"$${VERSION}\\\"
 DEFINES += APP_BUILD=\\\"$${BUILD}\\\"
+
+FONTTEST = $$(WITH_FONTS)
+isEmpty( FONTTEST ) | !equals($$upper(FONTTEST), "NO" ) {
+# Include the font resources (by defining "INCLUDE_FONTS", like a
+# "#define INCLUDE_FONTS" C preprocessor) if WITH_FONTS is NOT defined in the
+# environment or is NOT set to a case insensitive "no"
```

Comment:
This is confusing - a) are fonts included by default? (they should be) and b) how can one disable them easily?

Reply:
A). Yes
B). In a *nix shell set the environmental variable `WITH_FONTS` to (case insensitive) `no`; other OS have other comparable means. Typically this means doing something like

   `export WITH_FONTS="no"`

before running `qmake` (or once I do the __next__ commit `cmake`)...

Recent Qt Creator versions allow manipulation of variables like this for builds done within themselves:

![screenshot_20171206_155356](https://user-images.githubusercontent.com/6163092/33670757-b5b3ba0c-da9d-11e7-8f95-da0ec471e406.png)

_Notice that there is a summary at the top of the selection which indicates what Qt is "doctoring" for the build environment (and you can see I am also tweaking a couple of my locale settings) - there is a similar section (in the Qt Kit "Run" part, the above screen is the "Build" one) for when running the application from within the IDE - that is __very__ useful for testing the I18n GUI stuff for the "auto" locale choice as it means I do not have to reset my whole desktop to French/Russian/Chinese to check that that option will work..._ 🤓 😀 

The reason I strongly urge us to support environmental variable controls for build options is that there is no need to modify the project files (and thus tickle `git`) to change the configuration of what gets built - which makes things nicer for the packagers such as @csmall .

Sadly I had to abandon some project file stuff to download the font files from the Ubuntu server __only if needed__ because although I had something that would work for Linux (with `wget` + `tar`) I could not manufacture something we could rely on for macOs & 'Doze and I would need something to cover __all__ of them before I cut the new Font files from being directly included in the source as they are with this (and @zzbazinga 's) original attempt.

Such environmental variables are also something that can be (possibly on a temporary basis) varied in the CI build matrix for testing - although to keep the standard test CI jobs per build count down, this would want to be reset before final commitment to the development branch.

Reply:
OK, how about this comment then, I think it will be clearer:

```
Do export WITH_FONTS="no" to disable...
```

Ehh wait, what are we disabling? Why not just load fonts if they are available and don't load them if they're not?

Reply:
Because they are currently included in the source (I would like them not to be but I cannot make that so at the moment) but packagers (or those who have the fonts already in their systems) will not want to bloat the executable with them or **do not want them included in binaries that have licensing conditions that prevent the fonts from being included in their distributions**.

Reply:
tl;dr; Mudlet is in the "main" section for Debian - including the Ubuntu fonts in the binary package means it becomes, I believe, "non-free" see https://www.debian.org/distrib/packages#note ...